bin/templates/scripts/cordova/lib/clean.js

```diff
@@ -24,19 +24,26 @@ const superspawn = require('cordova-common').superspawn;
 
 const projectPath = path.join(__dirname, '..', '..');
 
-module.exports.run = function () {
-    const projectName = shell.ls(projectPath).filter(function (name) {
-        return path.extname(name) === '.xcodeproj';
-    })[0];
+module.exports.run = () => {
+    const projectName = shell.ls(projectPath).filter(name => path.extname(name) === '.xcodeproj')[0];
 
     if (!projectName) {
         return Q.reject('No Xcode project found in ' + projectPath);
     }
 
-    return superspawn.spawn('xcodebuild', ['-project', projectName, '-configuration', 'Debug', '-alltargets', 'clean'], { cwd: projectPath, printCommand: true, stdio: 'inherit' })
-        .then(function () {
-            return superspawn.spawn('xcodebuild', ['-project', projectName, '-configuration', 'Release', '-alltargets', 'clean'], { cwd: projectPath, printCommand: true, stdio: 'inherit' });
-        }).then(function () {
-            return shell.rm('-rf', path.join(projectPath, 'build'));
-        });
+    return superspawn.spawn(
+        'xcodebuild',
+        ['-project', projectName, '-configuration', 'Debug', '-alltargets', 'clean'],
+        { cwd: projectPath, printCommand: true, stdio: 'inherit' }
+    )
+        .then(
+            () => superspawn.spawn(
+                'xcodebuild',
+                ['-project', projectName, '-configuration', 'Release', '-alltargets', 'clean'],
+                { cwd: projectPath, printCommand: true, stdio: 'inherit' }
+            )
+        )
```

Comment:
Ugh, can we please transform this to something like the following:
```js
    const xcodebuildClean = configName => {
        return superspawn.spawn(
            'xcodebuild',
            ['-project', projectName, '-configuration', configName, '-alltargets', 'clean'],
            { cwd: projectPath, printCommand: true, stdio: 'inherit' }
        );
    };

    return xcodebuildClean('Debug')
        .then(() => xcodebuildClean('Release'))
```

If it's too much out of scope for this PR that's OK, but I hope the new formatting isn't here to stay :sweat_smile:

Reply:
def. out of scope.